src/app/models/AudioEventTag.ts

```
@@ -0,0 +1,46 @@
+import {
+  DateTimeTimezone,
+  dateTimeTimezone,
+  Id,
+} from "@interfaces/apiInterfaces";
+import { AbstractModel } from "./AbstractModel";
+
+export interface IAudioEventTag {
+  id?: Id;
+  audioEventId?: Id;
+  tagId?: Id;
+  creatorId?: Id;
+  updaterId?: Id;
+  createdAt?: DateTimeTimezone | string;
+  updatedAt?: DateTimeTimezone | string;
+}
+
+export class AudioEventTag extends AbstractModel implements IAudioEventTag {
+  public readonly kind: "AudioEventTag" = "AudioEventTag";
+  public readonly id?: Id;
+  public readonly audioEventId?: Id;
+  public readonly tagId?: Id;
+  public readonly creatorId?: Id;
+  public readonly updaterId?: Id;
+  public readonly createdAt?: DateTimeTimezone;
+  public readonly updatedAt?: DateTimeTimezone;
+
+  constructor(audioEventTag: IAudioEventTag) {
+    super(audioEventTag);
+
+    this.createdAt = dateTimeTimezone(audioEventTag.createdAt as string);
+    this.updatedAt = dateTimeTimezone(audioEventTag.updatedAt as string);
+  }
+
+  public toJSON() {
+    return {
+      id: this.id,
+      audioEventId: this.audioEventId,
+      tagId: this.tagId,
+    };
+  }
+
+  public get viewUrl(): string {
+    return "/BROKEN_LINK";
+  }
```

leave a note: this will probably never have it's own view. Should return `null`?

---

Ill throw an error instead of `null`, it'll be caught easier if a view is ever made.